src/log-store/src/kafka/log_store.rs

```diff
@@ -48,33 +115,103 @@ impl LogStore for KafkaLogStore {
         EntryImpl::new(data.as_ref().to_vec(), entry_id, ns)
     }
 
-    /// Append an `Entry` to WAL with given namespace and return append response containing
-    /// the entry id.
+    /// Appends an entry to the log store and returns a response containing the entry id and an optional entry offset.
     async fn append(&self, entry: Self::Entry) -> Result<AppendResponse> {
-        todo!()
+        let entry_id = entry.id;
+        let topic = entry.ns.topic.clone();
+
+        let offset = self.append_batch_to_topic(vec![entry], topic).await?.1;
+        Ok(AppendResponse {
+            entry_id,
+            offset: Some(offset),
+        })
     }
 
-    /// For a batch of log entries belonging to multiple regions, each assigned to a specific topic,
-    /// we need to determine the minimum log offset returned for each region in this batch.
-    /// During replay, we use this offset to fetch log entries for a region from its assigned topic.
-    /// After fetching, we filter the entries to obtain log entries relevant to that specific region.
+    /// For a batch of entries belonging to multiple regions, each assigned to a specific topic,
+    /// we need to determine the minimum start offset returned for each region in this batch.
+    /// During replay a region, we use this offset to fetch entries for the region from its assigned topic.
+    /// After fetching, we filter the entries to obtain entries relevant to the region.
     async fn append_batch(&self, entries: Vec<Self::Entry>) -> Result<AppendBatchResponse> {
-        todo!()
+        if entries.is_empty() {
+            return Ok(AppendBatchResponse::default());
+        }
+
+        // The entries are grouped by topic since the number of regions might be very large
+        // while the number of topics are well controlled,
+        let mut topic_entries: HashMap<_, Vec<_>> = HashMap::new();
+        // A utility map used to construct the result response.
+        let mut topic_regions: HashMap<_, Vec<_>> = HashMap::new();
+        for entry in entries {
+            let topic = entry.ns.topic.clone();
+            let region_id = entry.ns.region_id;
+            topic_entries.entry(topic.clone()).or_default().push(entry);
+            topic_regions.entry(topic).or_default().push(region_id);
+        }
+
+        // Appends each group of entries to the corresponding topic.
+        let tasks = topic_entries
+            .into_iter()
+            .map(|(topic, entries)| self.append_batch_to_topic(entries, topic))
```

Comment:
If all `entries` passed `append_batch_to_topic` to are the same(what you did here), we can change the code to: 

```rust

// Let's group by the entries by `region_id` first (Same as by `topic`, implicitly)

let tasks = ...
    self.append_batch_to_topic(entries)

//  self.append_batch_to_topic(entries) returns `(RegionId, Offest)`
```

Then, we can remove the above mapping.

Reply:
Writing to kafka will be refactored. See the commits to be pushed soon.